src/gpuarray_buffer_cuda.c

```diff
@@ -637,6 +643,7 @@ static int cuda_records(gpudata *a, int flags, CUstream s) {
   if (flags & CUDA_WAIT_WRITE)
     a->ctx->err = cuEventRecord(a->wev, s);
   cuda_exit(a->ctx);
+  a->ls = s;
```

Comment:
Normally wait/record calls should go in pairs, so the logic would be fine.  Of course if that is not respected, then there could be problems with the shortcut logic.

It's true that if there are no flags set, then we should probably not record the stream.